Composer/packages/client/src/pages/design/DebugPanel/TabExtensions/DiagnosticsTab/DiagnosticsStatus.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import { useRecoilState } from 'recoil';
+
+import { debugPanelExpansionState } from '../../../../../recoilModel';
+
+import { useDiagnosticsStatistics } from './useDiagnostics';
+
+/**
+ * Displays how many errors and warnings in current project.
+ */
+export const DiagnosticsStatus = () => {
+  const [, setExpansion] = useRecoilState(debugPanelExpansionState);
```

Comment:
useRecoilSetState?

Reply:
nit: We are trying to centralize side effects through dispatchers.

Reply:
updated to useSetRecoilState.

Reply:
I found many appStates constants are not exposed by dispatcher... so in this PR i just keep the boolean state and target to optimize it in the future